### PR TITLE
feat: harden checkout

### PR DESCRIPTION
**Pull request by laurentsimon**
See https://github.com/slsa-framework/slsa-github-generator/issues/626#issuecomment-1238324548

### CODE REVIEW

File: .github/actions/checkout-go/action.yml [hunk 1]
```diff
@@ -55,6 +56,9 @@ runs:
         repository: "${{ inputs.repository }}"
         token: "${{ inputs.token }}"
 
+    - name: Verify checkout
+      uses: slsa-framework/slsa-github-generator/.github/actions/verify-checkout@main
```
Comment (laurentsimon) [inline .github/actions/checkout-go/action.yml:60]: Once this is merged, I will update this code to a hash pin instead of "@main"

**Review by joshuagl — APPROVED** [review]

**Comment by laurentsimon** [comment]
@naveensrinivasan any idea why the `compute-sha256` needed to be re-compiled?

**Comment by ianlewis** [comment]
> @naveensrinivasan any idea why the `compute-sha256` needed to be re-compiled?

That's my fault. That check was failing but #646 got auto-merged because the check isn't a required check on the protected branch.

**Comment by laurentsimon** [comment]
no problem. I understand. ncc got updated and that's why it failed. Thanks

**Comment by ianlewis** [comment]
I added the check-dist as a required check so auto-submits won't work if it fails.

**Pull request merged**